### PR TITLE
Stp documentation

### PR DESCRIPTION
**Pull request by hilmarm**
This is #68 with a few additions.

### CODE REVIEW

File: network_configuration/stp.md [hunk 1]
```diff
@@ -96,7 +116,7 @@ The following steps of configuring the ports and attaching them to the bridge ca
 
 After setting up the bridge and bridge ports, the RSTP state on the bridge can be managed over the `mstpdctl` utility. The following command shows an example command output.
 
-**WARNING**: The `brctl` tool does not work with bridges managed by `mstpd`.
+**WARNING**: The `brctl` tool does not work with bridges managed by `mstpd`. For a bridge control tool use `mstpctl` instead.
```
Comment (rr-santos) [inline network_configuration/stp.md:119]:
```suggestion
**WARNING**: The `brctl` tool does not work with bridges managed by `mstpd`. Use `mstpctl` instead.
```

File: network_configuration/stp.md [hunk 2]
```diff
@@ -24,6 +24,26 @@ An STP-enabled bridge can be created using iproute2 with the following command:
...
+```
+
+```
+10-swbridge.network:
+
+[Match]
+Name=swbridge
+```
+
 The following steps of configuring the ports and attaching them to the bridge can be seen in [VLAN Bridging](network_configuration/vlan_bridging.html#vlan-bridging-8021q).
```
Comment (rr-santos) [inline network_configuration/stp.md:47]:
```suggestion
The following steps of configuring the ports and attaching them to the bridge can be seen in [VLAN Bridging](https://docs.bisdn.de/network_configuration/vlan_bridging.html#vlan-bridging-8021q).
```
(the original link did not work for me here and in the other links in this page)

**Review by rr-santos — COMMENTED** [review]
See comment about the links in this page

File: network_configuration/stp.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+## Introduction
+
+The Spanning Tree Protocol (STP) is a protocol meant to build loop-less topologies in Layer 2 networks. It works by distributively creating network paths without loops that could be harmful in case of e.g. flooding broadcast messages, by disabling (blocking) ports in the configured bridges. This document shows how to configure and test the STP implementation available in Linux.
+
+**WARNING**: BISDN Linux currently supports standard STP (IEEE Standard 802.1d) and RSTP (IEEE Standard 802.1w). For current network topologies, RSTP is the protocol more commonly used. For RSTP you *have* to use the packaged [mstpd](https://github.com/mstpd/mstpd).
+{: .label .label-yellow }
+
+## STP Configuration Instructions
+
+For configuring STP we assume the following configuration. In this document only the switch configuration is shown.
```
Comment (jklare) [inline network_configuration/stp.md:17]:
```suggestion
The STP configuration examples below assume the following topology. Throughout all examples given here, only the switch configuration side is shown.
```

File: network_configuration/stp.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+# Rapid Spanning Tree (RSTP)
+
+## Introduction 
+
+Meant as an improvement to the original STP standard, RSTP improves Spanning Tree convergence after network topology changes. If the switch is configured for RSTP and receives a STP (802.1d) BPDU, then STP functionality is assumed.
+
+There is currently no RSTP support in the Linux Kernel and therefore BISDN Linux uses [mstpd](https://github.com/mstpd/mstpd) to configure and manage RSTP.
+
+mstpd is managed by systemd and is disabled by default. For documentation on how to manage the service refer to [systemd getting started](https://docs.bisdn.de/getting_started/configure_baseboxd.html#getting-started).
```
Comment (jklare) [inline network_configuration/stp.md:102]:
```suggestion
mstpd is managed by systemd and is disabled by default. For documentation on how to manage systemd services please refer to [systemd getting started](https://docs.bisdn.de/getting_started/configure_baseboxd.html#getting-started).
```

File: network_configuration/stp.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+```
+
+```
+10-swbridge.network:
+
+[Match]
+Name=swbridge
+```
+
+The following steps of configuring the ports and attaching them to the bridge can be seen in [VLAN Bridging](network_configuration/vlan_bridging.html#vlan-bridging-8021q).
```
Comment (jklare) [inline network_configuration/stp.md:47]:
```suggestion
The necessary commands for configuring and attaching the ports to the bridge are documented here: [VLAN Bridging](network_configuration/vlan_bridging.html#vlan-bridging-8021q).
```

File: network_configuration/stp.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+## RSTP configuration
+
+When `mstpd` is enabled, any configured bridge will have STP enabled by default. Therefore, it is not required to specify the `stp_state` flag when creating a bridge:
+
+```
+ip link add name swbridge type bridge vlan_filtering 1
+```
+
+
+The following steps of configuring the ports and attaching them to the bridge can be seen in [VLAN Bridging](network_configuration/vlan_bridging.html#vlan-bridging-8021q).
```
Comment (jklare) [inline network_configuration/stp.md:113]:
```suggestion
The necessary commands for configuring and attaching the ports to the bridge are documented here:  [VLAN Bridging](network_configuration/vlan_bridging.html#vlan-bridging-8021q).
```

File: network_configuration/stp.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+## RSTP operation
+
+After setting up the bridge and bridge ports, the RSTP state on the bridge can be managed over the `mstpdctl` utility. The following command shows an example command output.
```
Comment (jklare) [inline network_configuration/stp.md:117]:
```suggestion
After setting up the bridge and its ports, the RSTP state on the bridge can be managed with the `mstpdctl` utility. The following command shows an example command output.
```

File: network_configuration/stp.md (hunk 2 above)
Comment (hilmarm) [reply network_configuration/stp.md:47]: fixed. Refering relatively is fine when it's done right, which it is not here. Here we would have to remove the network_configuration directory, as we are already in it. Absolute path is clear.

File: network_configuration/stp.md [hunk 8]
```diff
@@ -0,0 +1,149 @@
+---
+title: Spanning Tree Protocol (STP)
+parent: Network Configuration
+---
+
+# Spanning Tree Protocol (STP)
+
+## Introduction
+
+The Spanning Tree Protocol (STP) is a protocol meant to build loop-less topologies in Layer 2 networks. It works by distributively creating network paths without loops that could be harmful in case of e.g. flooding broadcast messages, by disabling (blocking) ports in the configured bridges. This document shows how to configure and test the STP implementation available in Linux.
```
[the remark this answers is not in the document]
Comment (hilmarm) [reply network_configuration/stp.md:10]: fixed

File: network_configuration/stp.md (hunk 3 above)
Comment (hilmarm) [reply network_configuration/stp.md:17]: sounds better.

File: network_configuration/stp.md (hunk 4 above)
Comment (hilmarm) [reply network_configuration/stp.md:102]: fixed

File: network_configuration/stp.md (hunk 5 above)
Comment (hilmarm) [reply network_configuration/stp.md:47]: Also flows much better. Added.

File: network_configuration/stp.md (hunk 7 above)
Comment (hilmarm) [reply network_configuration/stp.md:117]: good catch. fixed

File: network_configuration/stp.md [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+
+**WARNING**: BISDN Linux currently supports standard STP (IEEE Standard 802.1d) and RSTP (IEEE Standard 802.1w). For current network topologies, RSTP is the protocol more commonly used. For RSTP you *have* to use the packaged [mstpd](https://github.com/mstpd/mstpd).
```
[the remark this answers is not in the document]
Comment (hilmarm) [reply network_configuration/stp.md:12]: Better wording, added.

**Comment by jklare** [comment]
lgtm

File: network_configuration/stp.md [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+## RSTP configuration
+
+When `mstpd` is enabled, any configured bridge will have STP enabled by default. Therefore, it is not required to specify the `stp_state` flag when creating a bridge:
+
+```
+ip link add name swbridge type bridge vlan_filtering 1
+```
+
+
+The following steps of configuring the ports and attaching them to the bridge can be seen in [VLAN Bridging](network_configuration/vlan_bridging.html#vlan-bridging-8021q).
```
Comment (rr-santos) [inline network_configuration/stp.md:113]:
```suggestion
The following steps of configuring the ports and attaching them to the bridge can be seen in [VLAN Bridging](https://docs.bisdn.de/network_configuration/vlan_bridging.html#vlan-bridging-8021q).
```

**Review by rr-santos — COMMENTED** [review]
Other than the last link that needs to be fixed, LGTM

File: network_configuration/stp.md [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+```
+
+```
+10-swbridge.network:
+
+[Match]
+Name=swbridge
+```
+
+The necessary commands for configuring and attaching the ports to the bridge are documented here: [VLAN Bridging](https://docs.bisdn.de/network_configuration/vlan_bridging.html#vlan-bridging-8021q).
```
Comment (jklare) [inline network_configuration/stp.md:47]: why do we need the absolute link path here? we are using internal links throughout all the docs, so i why not here?

**Comment by rr-santos** [comment]
LGTM

File: network_configuration/stp.md [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+## RSTP configuration
+
+When `mstpd` is enabled, any configured bridge will have STP enabled by default. Therefore, it is not required to specify the `stp_state` flag when creating a bridge:
+
+```
+ip link add name swbridge type bridge vlan_filtering 1
+```
+
+
+The following steps of configuring the ports and attaching them to the bridge can be seen in [VLAN Bridging](https://docs.bisdn.de/network_configuration/vlan_bridging.html#vlan-bridging-8021q).
```
Comment (jklare) [inline network_configuration/stp.md:113]: why do we need the absolute link path here? we are using internal links throughout all the docs, so i why not here?

File: network_configuration/stp.md (hunk 11 above)
Comment (hilmarm) [reply network_configuration/stp.md:47]: It was not consistently used. How do I link internally if I want to go up one directory? i.e. ../getting_started/something.md

Comment (hilmarm) [reply network_configuration/stp.md:47]: Although here we can just point to the file in the same relative directory, network_configuration

Comment (hilmarm) [reply network_configuration/stp.md]: fixed

File: network_configuration/stp.md (hunk 12 above)
Comment (hilmarm) [reply network_configuration/stp.md:113]: Changed link type

File: network_configuration/stp.md [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
```
Comment (rubensfig) [inline network_configuration/stp.md:113]:
```suggestion
The following steps of configuring the ports and attaching them to the bridge can be seen in [VLAN Bridging](/network_configuration/vlan_bridging.html#vlan-bridging-8021q).
```

**Pull request merged**